text_sampling/text_sampling.py

```diff
@@ -0,0 +1,271 @@
+        if template_idx is None:
+            prompt_template = self.rnd_prompt_templates()
+        else:
+            prompt_template = self.prompt_templates[template_idx]
+        return prompt_template
+
+    def sample(self, sample, template_idx=None):
+        prompt_template = self.get_prompt_template_from_template_idx(template_idx)
+        sample_dict = self.get_sample_dict(sample, prompt_template.template)
+        return prompt_template.insert(sample_dict)
+
+    def __getitem__(self, sample_idx, template_idx=None):
+        sample = self.df.iloc[sample_idx]
+        return self.sample(sample, template_idx)
+
+    def apply_sampling(self, template_idx=None):
+        self.df["sample"] = self.df.apply(
+            lambda sample: self.sample(sample, template_idx), axis=1
+        )
+
+    def export(self):
+        print_data = {
+            "split": [],
+            "rows": [],
+            "path": [],
+        }
+        for split in self.df.split.unique():
+            # subselect for split
+            df_out = self.df.copy()
+            df_out = df_out[df_out["split"] == split]
+            if self.benchmarking_templates:
+                df_out[["input", "output"]] = df_out["sample"].str.split(
+                    pat="<EOI>", n=1, expand=True
+                )
+                df_out.drop(
+                    [
+                        x
+                        for x in df_out.columns.tolist()
+                        if x not in ["input", "output"]
+                    ],
+                    axis=1,
+                    inplace=True,
+                )
+            else:
+                df_out.drop(
+                    [x for x in df_out.columns.tolist() if x != "sample"],
+                    axis=1,
+                    inplace=True,
+                )
+                df_out.columns = ["text"]
+
+            # save
+            if self.benchmarking_templates:
+                output_path = self.path_data_dir + f"/{split}_benchmark.jsonl"
+            else:
+                output_path = self.path_data_dir + f"/{split}.jsonl"
+            with open(output_path, "w") as f:
+                f.write(df_out.to_json(orient="records", lines=True, force_ascii=False))
+
+            # stats
+            rows_split = len(df_out)
+            print_data["split"].append(split)
+            print_data["rows"].append(rows_split)
+            print_data["path"].append(output_path)
+
+        print_data["split"].append("total")
+        print_data["rows"].append(len(self.df))
+        print_data["path"].append("")
+        return pd.DataFrame(print_data)
+
+    def apply_and_export(self, template_idx=None):
+        self.apply_sampling(template_idx)
+        df_results = self.export()
+        print(df_results.to_string())
+
+
+if __name__ == "__main__":
+    for path in glob.glob(__file__.replace("text_sampling/text_sampling.py", "data/*")):
+        path_meta = path + "/meta.yaml"
+        path_data = path + "/data_clean.csv"
```

Comment:
since we anyhow will only run on Linux this shouldn't be an issue?

Reply:
Will add this later for OS-proof paths:
> use something like os.path.join (or the equivalent function in pathlib) it takes care of that for you

Reply:
kept as it is as we will not run this on Windows